# chapter 17
# built in errors
# exception, how to handle them
# raise your own errors, debugging, etc etc

# syntax error 

# def func:
#     pass

# name = 'harshit'

# indentation error 

# def func():
#     print('hello')
#   print('world')

# name error

# print(func)

# print(5 * 'harshit')

# l = [1,2,3]
# print(l[4])

# value error 
# s = '123'
# print(int(s))

# attribute error

# l = [1,2,3]
# l.push('12')

# key error 
d = {'name': 'harshit'}
print(d['age'])